lib/src/macro.dart

```diff
@@ -4,44 +4,43 @@ import 'types.dart';
 
 abstract class Macro {}
 
-abstract class TypeMacro implements Macro {}
-
-abstract class DeclarationMacro implements Macro {}
-
-abstract class DefinitionMacro implements Macro {}
-
 abstract class ClassTypeMacro implements TypeMacro {
-  void type(TargetClassType type);
+  void type(ClassType type, TypeBuilder builder);
```

Comment:
I still don't like the `type` name, and we want it to be more specific to not have collisions as well with other `*TypeMacro`s.

What about `visitClassType`?